lib/elixir/lib/module/types/descr.ex

```diff
@@ -202,4 +230,82 @@ defmodule Module.Types.Descr do
         (mask &&& val) !== 0,
         do: {type, [], []}
   end
+
+  defp bitmap_count_top_types(val), do: Integer.digits(val, 2) |> Enum.sum()
+
+  ## Atoms
+
+  # The atom component of a type consists of pairs `{MapSet.t(atom()), boolean()}`.
+  # If the boolean is true, the type represents the union of the atoms in the set.
+  # If the boolean is false, the type represents all the possible atoms, except the ones
+  # in the set.
+  #
+  # Example:
+  #   - `{MapSet.new([:a, :b]), true}` represents type `:a or :b`
+  #   - `{MapSet.new([:c, :d]), false}` represents type `atom() \ (:c or :d)
+  #
+  # `{MapSet.new([]), false}` is the `atom()` top type, as it is the difference
+  # of `atom()` with an empty list.
+
+  defp atom_mk(a) when is_atom(a), do: {:union, MapSet.new([a])}
+  defp atom_mk(as) when is_list(as), do: {:union, MapSet.new(as)}
```

Comment:
Nitpick: we rarely use abbreviations in the Elixir codebase. Maybe we call it `atom_new`? Also, I would prefer to avoid the code from being too polymorphic. Should we enforce the input is always a list?

Reply:
Yes let's keep only the list input